File: solutions/ruby/raindrops/2/raindrops.rb
module Raindrops
  def self.convert(number)
    raindrops = { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }
    string = ''

    raindrops.each { |prime, sound| string += sound if (number % prime).zero? }

    string.empty? ? number.to_s : string
  end
end

File: solutions/python/raindrops/2/raindrops.py
""" Raindrops exercise """
def convert(number):
    words = {
        3: 'Pling',
        5: 'Plang',
        7: 'Plong'
    }

    return ''.join([
        word for num, word in words.items() if number % num == 0
    ]) or f"{number}"
